src/backend/access/aocs/aocssegfiles.c

```diff
@@ -454,6 +454,136 @@ GetAOCSSSegFilesTotals(Relation parentrel, Snapshot appendOnlyMetaDataSnapshot)
 	return totals;
 }
 
+/*
+ * GetAOCSSSegFilesTotals
+ *
+ * Get the total FileSegTotals information for a specific AO table
+ * from the pg_aoseg tables on the entire cluster.
+ * Note: result is palloced, caller should free it.
+ */
+FileSegTotals *GetAOCSSegFilesTotalsCluster(Relation parentrel)
+{
+	FileSegTotals *result;
+	StringInfoData sqlstmt;
+	Relation	aosegrel;
+	volatile bool connected = false;
+	Oid			segrelid = InvalidOid;
+
+	Assert(Gp_role == GP_ROLE_DISPATCH);
+	Assert(RelationIsAoCols(parentrel));
+
+	GetAppendOnlyEntryAuxOids(RelationGetRelid(parentrel), NULL,
+							  &segrelid, NULL, NULL, NULL, NULL);
+	Assert(OidIsValid(segrelid));
+
+	result = (FileSegTotals *) palloc0(sizeof(FileSegTotals));
+
+	/*
+	 * open the aoseg relation
+	 */
+	aosegrel = heap_open(segrelid, AccessShareLock);
+
+	initStringInfo(&sqlstmt);
+	appendStringInfo(&sqlstmt, "select tupcount, varblockcount, vpinfo "
+					 "from gp_dist_random('%s.%s')",
+					 get_namespace_name(RelationGetNamespace(aosegrel)),
+					 RelationGetRelationName(aosegrel));
+
+	heap_close(aosegrel, AccessShareLock);
+
+	/*
+	 * Temporarily disable ORCA because it's slow to start up, and it
+	 * wouldn't come up with any better plan for the simple queries that
+	 * we run.
+	 * Plus this function may be called during Orca's work, and that will
+	 * result in nested Orca planning, which is not supported.
+	 */
+	bool save_optimizer_guc_value = optimizer;
+	optimizer = false;
+
+	PG_TRY();
+	{
+		if (SPI_OK_CONNECT != SPI_connect())
+			ereport(ERROR,
+					(errcode(ERRCODE_INTERNAL_ERROR),
+					 errmsg("unable to obtain AO relation information from segment databases"),
+					 errdetail("SPI_connect failed in GetAOCSSegFilesTotalsCluster")));
+
+		connected = true;
+
+		if ((SPI_execute(sqlstmt.data, false, 0) <= 0) || (SPI_tuptable == NULL))
```

Comment:
This lines exceed 80 symbols border. I suggest to remove redundant parenthesis
```suggestion
		if (SPI_execute(sqlstmt.data, false, 0) <= 0 || SPI_tuptable == NULL)
```

Reply:
I've split into 2 lines. I prefer to use parenthesis. It is highly recommended by many recognized standards for mission critical systems (for ex. MISRA C (2012 Rule 12.1), CERT C (Rec. EXP00-C)).